uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/work-pattern-api.js

```diff
@@ -13,7 +13,43 @@ define([
     return api.extend({
 
       /**
-       * This method returns the calendar for the given contact(s) and period,
+       * Assigns a work pattern to a contact
+       *
+       * @param {string} contactId
+       * @param {string} workPatternID
+       * @param {string} effectiveDate
+       * @param {string} effectiveEndDate
+       * @param {string} changeReason
+       * @param {object} params - additional parameters
+       * @return {Promise} Resolved with {Array} All Work Patterns
+       */
+      assignWorkPattern: function (contactId, workPatternID, effectiveDate, effectiveEndDate, changeReason, params) {
+        return this.sendGET('ContactWorkPattern', 'create', _.assign({}, params, {
+          contact_id: contactId,
+          pattern_id: workPatternID,
+          effective_date: effectiveDate,
+          effective_end_date: effectiveEndDate,
+          change_reason: changeReason
+        })).then(function (data) {
+          return data.values;
+        });
+      },
+
+      /**
+       * Returns all the work patterns
+       *
+       * @param {object} params additional parameters
+       * @return {Promise} Resolved with {Array} All Work Patterns
+       */
+      get: function (params) {
+        return this.sendGET('WorkPattern', 'get', params || {})
```

Comment:
`_.assign` is necessary for other API calls because `params`and the other parameters needs to be merged into 1 object, but in this case there is no other parameters as the function has a single param, so there is no need to use `_.assign`